src/main/java/duke/command/ByeCommand.java

```diff
@@ -0,0 +1,29 @@
+package duke.command;
+
+import duke.exception.DukeException;
+
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+/**
+ * Command that contains details when user inputs Bye
+ */
+public class ByeCommand extends Command{
+    /**
+     * Basic Constructor
+     *
+     * @param storage Storage object to save
+     * @param taskList Tasklist to add task to
+     * @param ui Ui to display msg
+     */
+    public ByeCommand(Storage storage, TaskList taskList,Ui ui){
+        super(storage, taskList, ui);
+    }
+
+    @Override
+    public boolean exec() throws DukeException {
+        ui.goodbye();
+        return false;
```

Comment:
Can I ask why did you choose to make the execute function return the boolean instead of setting it as a parameter?

Reply:
Yup, I agree. Didn't think about it at that time. Thanks for the suggestion!